org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterColorColor.java

```diff
@@ -452,15 +457,21 @@ private void updateBrightness(PercentType brightness) {
         HSBType oldHSB = lastHSB;
         HSBType newHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), brightness);
         lastHSB = newHSB;
-        updateChannelState(newHSB);
+
+        if (currentOnOffState.get()) {
+            updateChannelState(newHSB);
+        }
     }
 
     private void updateColorHSB(DecimalType hue, PercentType saturation) {
         // Extra temp variable to avoid thread sync concurrency issues on lastHSB
         HSBType oldHSB = lastHSB;
         HSBType newHSB = new HSBType(hue, saturation, oldHSB.getBrightness());
         lastHSB = newHSB;
-        updateChannelState(newHSB);
+
```

Comment:
Ditto..

Probably you should also update the HSB when the device is off.

Instead of this..
```
HSBType newHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), oldHSB.getBrightness());
```
Try this..
```
HSBType newHSB = new HSBType(oldHSB.getHue(), oldHSB.getSaturation(), currentOnOffState.get() ? oldHSB.getBrightness() :0);
```

Reply:
But I must admit that my proposal does therefore mean that the brightness is not persisted across an on/off/on toggle. Hmm.

Reply:
So probably you need to first set `lastHSB` to persist the actual brightness. And after that you can call `updateChannelState` with the modified brightness from the ternary operator.

Reply:
I guess it's firstly important to note that this bug was introduced in #855. Prior to that this worked fine (or at least acceptably). My focus here is not to change the logic, but to fix the regression introduced in #855. This converter went through a lot of review by a number of people when it was written and I'd prefer not to change logic in this regression fix.

> Probably you should also update the HSB when the device is off.

The HSB is updated locally - I guess you mean it should be updated in OH. Personally, I would avoid this in this PR. It will be a change in functionality that could be annoying or confusing, and ultimately it doesn't actually matter if the device is not on.

Reply:
Ok.